torch/_inductor/codegen/triton.py

```diff
@@ -2115,6 +2115,79 @@ def codegen_node_schedule(self, node_schedule, numel, reduction_numel):
         src_code = kernel.codegen_kernel()
         kernel_name = self.define_kernel(src_code, node_schedule)
 
+        ################## auto tuning training data dump
+        log.warning("Kernel path : " + self.kernel_path_)
+        tuning_metadata_path = self.kernel_path_ + ".pkl"
+        import pickle, os
+        from ..utils import get_dtype_size
+
+        if not os.path.exists(os.path.dirname(tuning_metadata_path)):
+            os.makedirs(os.path.dirname(tuning_metadata_path))
+        with open(tuning_metadata_path, "wb") as f:
+            log.warning("Metadata path : " + tuning_metadata_path)
+            if len(nodes) == 1:
+                node = nodes[0]
+            else:
+                node = scheduler.FusedSchedulerNode(self.scheduler, nodes)
+
+            def get_read_write_removed(node):
+                if isinstance(node, scheduler.NopKernelSchedulerNode):
+                    return 0
+                reads = {dep.name for dep in node.read_writes.reads}
+                writes = {dep.name for dep in node.read_writes.writes}
+
+                def is_materialized(buf):
```

Comment:
I believe `V.wrapper` has a set of removed buffer names you might be able to use instead here.